### ISSUE TITLE
Mozilla removed the extension for Firefox.

### ISSUE THREAD

**Issue by Immortal-Sty** [omnivore-app/omnivore#3100]
We can't install browser extensions for Firefox, because Mozilla removed it.

**Comment by avelino** [comment]
https://addons.mozilla.org/en-US/firefox/addon/omnivore/

> ![Screenshot 2023-11-13 at 13 04 13](https://github.com/omnivore-app/omnivore/assets/31996/830db355-d846-4dc1-952c-fa55326d1609)

**Comment by jacksonh** [comment]
We've resubmitted to Mozilla. There was an issue with our source package that prevented them from building the extension from source. Hoping to get an update today.

**Comment by Harmonia2Side** [comment]
Oh no, just at the exact week I was about to start using it. Hope Mozilla fixes this soon

**Comment by jacksonh** [comment]
Latest update from Mozilla:

```
This add-on didn't pass review because of the following problems:



1) For add-ons that collect or transmit user data, the user must be informed and provided with a clear and easy way to control this data collection. The control mechanism must be shown at first-run of the add-on. The control should contain a choice accompanied by the data collection summary. Depending on the type of data being collected, the choice to send cannot be enabled by default. For the exact requirements, refer to https://extensionworkshop.com/documentation/publish/add-on-policies/#data-disclosure-collection-and-management . For an example of how to provide a consent and control dialog, see https://extensionworkshop.com/documentation/develop/best-practices-for-collecting-user-data-consents/ .

Also, if your add-on is listed on addons.mozilla.org, the listing needs to include a privacy policy, and a summary of the data collection should be mentioned in the add-on description.



2) Collecting personal or potentially identifiable user data like the tab url requires affirmative user consent (i.e. explicit opt-in from the user). It must be clear to the user that they give consent to the collection of personal data.

Please see https://extensionworkshop.com/documentation/develop/best-practices-for-collecting-user-data-consents/ for an example of how to present the affirmative consent. Also, a summary of the data collection must be added to the add-on description and in the text accompanying the consent.

For more information, refer to https://extensionworkshop.com/documentation/publish/add-on-policies/#data-disclosure-collection-and-management .



3) Please provide us with detailed information on how to test this add-on. If authentication to a website is necessary, give us a test username and password to facilitate our work. This can be provided in the Whiteboard field, which can be found in the Edit Listing page under the Technical Details section. This information is only available to reviewers.
```

**Comment by jacksonh** [comment]
In the meantime I'd suggest using this community plugin: https://addons.mozilla.org/en-US/firefox/addon/omnivore-list-popup/

**Comment by jacksonh** [comment]
Most of the changes done here, unfortunately (ironically) I had to poke some holes in the security policy to do this. Doesn't seem worth it to display a privacy policy, so will try to rework before submitting: https://github.com/omnivore-app/omnivore/pull/3137

**Comment by mrinc** [comment]
Been there and had the same fun :D  

The big notes: 
Person has to enter API key to have extension work - show a link to your PP there.  
Nothing in the extension should do anything until that is entered.  
That allows the person to decide if they want to use it or not.  
Also specifically list this as an extension only for omnivore, so people have already accepted your main PP before using this extension.   
As long as you aren't snapping logs/stats the above should suffice.  

As for testing, what you can do is add a fake API key - like ```AAAAAAAAAAAAAAAA-AA``` and just have the HTTP request handler ignore requests with an HTTP200 or have your server just return OK if that API key is received.  

When you pub source, just make sure they could re-build from that source and get the exact same output as the extension code being published... I've had fun with this one with a bunch of back and forth.   
I found the best way was to build my extension in a docker container that is controlled, and when they receive the source and try rebuild - it's 100% the same - rather than maybe they are 1 micro version diff on a lib or something and it caused a diff hash on the output files -_-

**Comment by jacksonh** [comment]
Yeah ours builds in a docker container that matches the config but it looks like we even have to guess which shell the reviewer is using. Latest issue is they don't have `popd` on their machine.

**Comment by mrinc** [comment]
@jacksonh LOL ... yea ... I feel you.  

Ref: [https://github.com/mrinc/F-B-Hide-Recommendations-and-Reels/blob/master/build.sh](https://github.com/mrinc/F-B-Hide-Recommendations-and-Reels/blob/master/build.sh) - this is my plugin where i had the same issues - its the build script.  

What I did was go straight into docker, no if's, but's, why's or who's - I have everything run from the container rather.  
That way I can guarantee I wont have issues with the reviewers or for a rebuild later on and also I don't have or don't expect any requirements on the host except for docker and docker compose - the rest is irel.... helps a lot.

If you need to, you can use my code as a base - I don't have issues with FF/Google for publishes anymore.  
The build proc is very simple, but consistent and repeatable.

**Comment by jacksonh** [comment]
> @jacksonh LOL ... yea ... I feel you.
> 
> Ref: https://github.com/mrinc/F-B-Hide-Recommendations-and-Reels/blob/master/build.sh - this is my plugin where i had the same issues - its the build script.
> 
> What I did was go straight into docker, no if's, but's, why's or who's - I have everything run from the container rather. That way I can guarantee I wont have issues with the reviewers or for a rebuild later on and also I don't have or don't expect any requirements on the host except for docker and docker compose - the rest is irel.... helps a lot.
> 
> If you need to, you can use my code as a base - I don't have issues with FF/Google for publishes anymore. The build proc is very simple, but consistent and repeatable.

madness. I wonder what they are even using, its definitely not a clean docker image.

**Comment by mrinc** [comment]
I cannot find any of the historic images that i was sent but they did mention for my plugin they used: ```Ubuntu 20.04 using node  v14.17.0, npm 6.14.13```  

After a few more failed submissions of trying to get them to be able to build it ... I just went straight docker and completely isolated everything.  

Although it is chaos, it does lead way into controlling your build perfectly :)  
And then going forward you'll never have build, review or even re-build issues... so worth it in the long run anyway.

**Comment by mrinc** [comment]
Also to note, the code scripts are *kinda* minified in src, ff may have an issue with that as well.  
It's not compiled, so should be fine ... but something to note as well which caused me some pain.  (I had tailwindcss included as a js file rather than compiling during build - and just removed it in order to get past verification)
